specs/guaranteed-gas-market.md

```diff
@@ -0,0 +1,137 @@
+# Guaranteed Gas Fee Market
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Gas Stipend](#gas-stipend)
+- [Limiting Guaranteed Gas](#limiting-guaranteed-gas)
+- [1559 Fee Market](#1559-fee-market)
+  - [Exponent Based Fee Reduction](#exponent-based-fee-reduction)
+- [Rationale for burning L1 Gas](#rationale-for-burning-l1-gas)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+[Deposited transaction](./glossary.md#deposited-transaction) are transactions on L2 that are
+initiated on L1. The gas that they use on L2 is bought on L1 via a gas burn or a direct payment. We
+maintain a fee market and hard cap on the amount of gas provided to all deposits in a single L1
+block.
+
+The gas provided to deposited transactions is sometimes called "guaranteed gas". The gas provided to
+deposited transactions is unique in the regard that it is not refundable. It cannot be refunded as
+it is sometimes paid for with a gas burn and there may not be any ETH left to refund.
+
+The **guaranteed gas** is composed of a gas stipend, and of any guaranteed gas the user would like
+to purchase (on L1) on top of that.
+
+Guaranteed gas on L2 is bought in the following manner. An L2 gas price is calculated via a 1559
+style gas market. The total amount of ETH required to buy that gas is then calculated
+(`guaranteed gas * L2 deposit basefee`). The contract then accepts that amount of ETH (in a future
+upgrade) or (only method right now), burns an amount of L1 gas that corresponds to the L2 cost
+(`L2 cost / L1 Basefee`).
+
+## Gas Stipend
+
+To offset the gas spent on the deposit event, we credit the amount of gas spent on the metering step
+times the current basefee to the cost of the L2 gas. The amount of gas is selected to represent the
+cost to the user. If the ETH price of the gas (gas times current L1 baseefee) is greater than the
+requested guaranteed gas times the L2 gas price, no L1 gas is burnt.
```

Comment:
s/spent on the metering step times the current basefee/spent on the metering step, multiplied by the current basefee, to the cost of the L2 gas/

It's not quite clear here what the "metering step" is, and in fact, I'm not sure it's what we want to say at all (the whole processing of the deposit is what we want to offset, right?).

Suggestion:

> To offset the gas spent on the deposit event, we credit `gas spent * L1 basefee` ETH to the cost of the L2 gas, where `gas spent` is the amount of L1 gas spent processing the deposit. If the ETH value of this credit is greater than the ETH value of the requested guaranteed gas (`requested guaranteed gas * L2 gas price`), no L1 gas is burnt.